src/azure-cli/azure/cli/command_modules/profile/tests/latest/test_subscription_selector.py

```diff
@@ -0,0 +1,127 @@
+
+
+# These dummy subscriptions are used in both unit test and interactive test
+DUMMY_SUBSCRIPTIONS = [
+    # 0: sub 2
+    {
+        "id": "00000000-0000-0000-0000-222222222222",
+        "name": "sub 2",
+        "tenantDefaultDomain": "tenant1.onmicrosoft.com",
+        "tenantDisplayName": "Tenant 1",
+        "tenantId": "00000000-0000-0000-1111-111111111111",
+        "environmentName": "AzureCloud",
+        "isDefault": True
+    },
+    # 1: sub 1
+    {
+        "id": "00000000-0000-0000-0000-111111111111",
+        "name": "SUB 1",
+        "tenantDefaultDomain": "tenant1.onmicrosoft.com",
+        "tenantDisplayName": "Tenant 1",
+        "tenantId": "00000000-0000-0000-1111-111111111111",
+        "environmentName": "AzureCloud",
+        "isDefault": False
+    },
+    # 2: sub 3
+    {
+        "id": "00000000-0000-0000-0000-333333333333",
+        "name": "Sub 3 with long long long long long name",
+        "tenantDefaultDomain": "tenant1.onmicrosoft.com",
+        "tenantDisplayName": "Tenant 1",
+        "tenantId": "00000000-0000-0000-1111-111111111111",
+        "environmentName": "AzureCloud",
+        "isDefault": False
+    },
+    # 3: tenant account
+    {
+        "id": "00000000-0000-0000-1111-222222222222",
+        "name": "N/A(tenant level account)",
+        "tenantDefaultDomain": "tenant2.onmicrosoft.com",
+        "tenantDisplayName": "Tenant 2",
+        "tenantId": "00000000-0000-0000-1111-222222222222",
+        "environmentName": "AzureCloud",
+        "isDefault": False
+    }
+]
+
+DUMMY_SUBSCRIPTIONS_NO_TENANT_INFO = [
+    {
+        "id": "00000000-0000-0000-0000-222222222222",
+        "name": "sub 2",
+        "tenantId": "00000000-0000-0000-1111-111111111111",
+        "environmentName": "AzureCloud",
+        "isDefault": True
+    },
+    {
+        "id": "00000000-0000-0000-0000-111111111111",
+        "name": "sub 1",
+        "tenantId": "00000000-0000-0000-1111-111111111111",
+        "environmentName": "AzureCloud",
+        "isDefault": False
+    }
+]
+
+
+class TestSubscriptionSelection(unittest.TestCase):
+
+    def setUp(self):
+        super().setUp()
+
+    def test_format_subscription_table(self):
+        sub = DUMMY_SUBSCRIPTIONS
+        from azure.cli.core.style import format_styled_text
+
+        # We have to force format_styled_text to use 'dark' theme, otherwise other tests in the same process
+        # may change format_styled_text.theme to 'none' and format_styled_text will not print color, leading to
+        # unmatched result. For example, this command fails:
+        #   azdev test test_account_show test_format_subscription_table
+        # create=True is required as running this test alone won't set format_styled_text.theme. patch.object
+        # will fail because of the missing theme attribute. For example, this command fails:
+        #   azdev test test_format_subscription_table
+        with mock.patch.object(format_styled_text, 'theme', 'dark', create=True):
```

Comment:
This mock can also be achieved by deleting the `theme` attribute, just like running `test_format_subscription_table` alone. Pytest's [`monkeypatch`](https://docs.pytest.org/en/latest/how-to/monkeypatch.html) fixture has a [`monkeypatch.delattr(obj, name, raising=True)`](https://docs.pytest.org/en/latest/reference/reference.html#pytest.MonkeyPatch.delattr) method for this purpose, but I can't find a way to do this with [`unittest.mock`](https://docs.python.org/3/library/unittest.mock.html).

Reply:
We can write some custom logic to achieve this:

```py
        is_attr_set = False
        original_attr_value = None
        if hasattr(format_styled_text, 'theme'):
            is_attr_set = True
            original_attr_value = format_styled_text.theme
            delattr(format_styled_text, 'theme')

        ...

        if is_attr_set:
            format_styled_text.theme = original_attr_value
```

or even wrap it as a context manager:

```py
class delattr_patch:
    def __init__(self, object, attr):
        self.object = object
        self.attr = attr
        self.is_attr_set = False
        self.original_attr_value = None

    def __enter__(self):
        if hasattr(self.object, self.attr):
            self.is_attr_set = True
            self.original_attr_value = getattr(self.object, self.attr)
            delattr(self.object, self.attr)

    def __exit__(self, exc_type, exc_value, traceback):
        if self.is_attr_set:
            setattr(self.object, self.attr, self.original_attr_value)
```

Then we can use it as 

```py
        with delattr_patch(format_styled_text, 'theme'):
            ...
```